src/containers/registration/hooks/useConsultingTypeAgencySelection.ts

```diff
@@ -0,0 +1,84 @@
+import { useState, useEffect } from 'react';
+import {
+	ConsultantDataInterface,
+	ConsultingTypeInterface,
+	AgencyDataInterface
+} from '../../../globalState';
+import { useAppConfig } from '../../../hooks/useAppConfig';
+
+export const useConsultingTypeAgencySelection = (
+	consultant: ConsultantDataInterface,
+	consultingType: ConsultingTypeInterface,
+	agency: AgencyDataInterface
+) => {
+	const settings = useAppConfig();
+	const [consultingTypes, setConsultingTypes] = useState<
+		ConsultingTypeInterface[]
+	>([]);
+	const [agencies, setAgencies] = useState<AgencyDataInterface[]>([]);
+
+	useEffect(() => {
+		if (!consultant) {
+			return;
+		}
+
+		// When we've the multi tenancy with single domain we can simply ignore the
+		// consulting types because we'll get agencies across tenants
+		if (
+			settings.multitenancyWithSingleDomainEnabled &&
+			consultant?.agencies?.length > 0
+		) {
+			setAgencies(consultant?.agencies);
+			setConsultingTypes([consultingType]);
+			return;
+		}
+
+		const consultingTypes = consultant.agencies
+			// Remove consultingType duplicates
+			.reduce((acc: ConsultingTypeInterface[], { consultingTypeRel }) => {
+				if (
+					!acc.find(
+						(consultingType) =>
+							consultingType.id === consultingTypeRel.id
+					)
+				) {
+					acc.push(consultingTypeRel);
```

Comment:
```suggestion
					return [...acc, consultingTypeRel];
```
I would reduce array manipulation to a minimum